crates/bevy_asset/src/asset_server.rs

```diff
@@ -93,6 +93,15 @@ impl AssetServer {
     }
 
     pub(crate) fn register_asset_type<T: Asset>(&self) -> Assets<T> {
+        if self
+            .server
+            .asset_lifecycles
+            .read()
+            .contains_key(&T::TYPE_UUID)
+        {
+            panic!("Error while registering new asset type. Asset with UUID: {:?} is already registered. Can not register another type with the same UUID",
+                T::TYPE_UUID);
```

Comment:
could you also log `std::any::type_name::<T>()` in the error?

Reply:
Ideally it would be possible to include the type name of the first type too, although I don't know if we store that information

Reply:
Added type name, but i don't think there is a way to get the original type. We only have trait object, so can't get the type from it.